js/reference.js

```diff
@@ -2838,11 +2838,13 @@
                 pseudoNameObj, pseudoName, isHash,
                 hasInbound, isEntity, hasPath, isEntityMode,
                 isEntry,
+                isCompactEntry,
                 colFks,
                 ignore, cols, col, fk, i, j;
 
             var context = this._context;
             isEntry = module._isEntryContext(context);
+            isCompactEntry = context.startsWith(module._contexts.COMPACT_ENTRY);
```

Comment:
You should add the `typeof context === "string"` before this. 

```javascript
var isCompactEntry = typeof context === "string" && context.startsWith(module._contexts.COMPACT_ENTRY);
```

As I mentioned, we support contextualized References and that's why some test cases are failing.